src/visualizers/SeqColor.ts

```diff
@@ -0,0 +1,258 @@
+import type p5 from 'p5'
+import {VisualizerDefault} from './VisualizerDefault'
+import type {VisualizerInterface} from '@/visualizers/VisualizerInterface'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import type {SequenceInterface} from '../sequences/SequenceInterface'
+
+const colorMap = new Map()
+let countClick = 0
+
+//Show description box by pressing left arrow key
+class SeqColor extends VisualizerDefault implements VisualizerInterface {
+    name = 'Sequence Color'
+    n = 40
+
+    params = {
+        n: {
+            value: this.n,
+            forceType: 'integer',
+            displayName: 'number of terms',
+            required: true,
+        },
+    }
+
+    private current_num = 0
+    private X = 0
+    private Y = 0
+    private subG = this.sketch.createGraphics(800, 90)
+    private boxIsShow = false
+    private primeNum: number[] = []
+    private count_prime = 0
+    private firstDraw = true
+
+    initialize(sketch: p5, seq: SequenceInterface) {
+        super.initialize(sketch, seq)
+        this.sketch = sketch
+        this.seq = seq
+        this.current_num = seq.first
+        this.X = 0
+        this.Y = 0
+        this.ready = true
+    }
+
+    setup() {
+        this.sketch.background('black')
+        this.sketch.colorMode(this.sketch.HSB, 360, 100, 100)
+        this.sketch.frameRate(30)
+
+        //Set position of the circle
+        this.X = 800 / this.n + 30
+        this.Y = 800 / this.n + 50
+
+        //Obtain all prime numbers from the sequence
+        for (let i = this.seq.first; i < this.n; i++) {
+            if (
+                this.isPrime(Number(this.seq.getElement(i))) == true
```

Comment:
You could store the result of `Number(this.seq.getElement(i))` in a variable and then re-use it on line 55 and line 57.